.changeset/wild-snakes-deliver.md

```diff
@@ -0,0 +1,45 @@
+---
+"thirdweb": minor
+---
+
+Add `valueERC20` option in `PreparedTransaction` object to specify the transaction cost in ERC20 tokens to allow opening the Pay Modal for ERC20 tokens if user does not have specified ERC20 tokens.
+
+```tsx
+const sendTx = useSendTransaction();
+
+return (
+  <button
+    onClick={async () => {
+      const someTx = { ... };
+
+      // declare that transaction cost is 10 USDC
+      someTx.valueERC20 = {
```

Comment:
can you actually mutate the transaction like that? i thought fields were read only

Reply:
Yes, that's why it's added on `PreparedTransaction` and not on `PrepareTransactionOptions`